osrf/vrx
# Buoyancy plugin refactor

This pull requests consolidates the two former buoyancy plugins (`buoyancy_gazebo_plugin` and `usv_gazebo_dynamics_plugin`) into a single one (`SimpleBuoyancy`). I'm open to suggestions for other names.

`SimpleBuoyancy` is based on `buoyancy_gazebo_plugin` with a few improvements and bug fixes from the previous version. Now, it's possible to specify via SDF multiple `buoyancy` blocks with the same `<link_name>` but different offsets. This is useful for the WAM-V as we want to specify the geometries of the hulls as offsets from `base_link`.

As an example, we can now reuse the same buoyancy plugin for buoys and for the WAM-V.

### How to test it?

Launch the simulation:

```
ros2 launch vrx_ros competition_local.launch.py ign_args:="-v 4 -r sydney_regatta.sdf"
```

Launch a WAM-V:

```
ros2 launch vrx_ign spawn.launch.py name:=wamv world:=sydney_regatta model:=wam-v x:=-532 y:=162 z:=0 R:=0 P:=0 Y:=1
```

You should see a reasonable floating WAM-V moving in sync with the waves. I also added to the scene a red buoy with the same plugin loaded. The floating behavior of the buoy should look OK as well.

### Bonus

This is a video of two WAM-Vs running the two buoyancy plugins considered: the `Surface` (based on the former `usv_gazebo_dynamics` and the `SimpleBuoyancy` introduced in this pull request. It's very qualitative but the behavior is quite similar.
![buoyancy_2](https://user-images.githubusercontent.com/1440739/186013773-6fd25271-4cab-4b4b-907c-14fb1d0552a1.gif)
